### PR TITLE
[portconfig]Auto FEC initial changes

### PR DESCRIPTION
**Pull request by dgsudharsan**
#### What I did
Initial changes for auto FEC. HLD https://github.com/sonic-net/SONiC/pull/1416

#### How I did it
Added default option of auto for FEC

#### How to verify it
Modified UT to verify

#### New command output (if the output of a command-line utility has changed)

```
admin@sonic:~$ show interfaces fec status
  Interface    FEC Oper    FEC Admin
  -----------  ----------  -----------
  Ethernet0         N/A           rs
 Ethernet32         N/A           rs
 Ethernet36         N/A          N/A
Ethernet112         N/A           rs
Ethernet116         N/A           rs
Ethernet120         N/A           rs
Ethernet124          rs         auto
```

### CODE REVIEW

**Comment by dgsudharsan** [comment]
/azpw.run Azure.sonic-utilities

**Comment by ridahanif96** [comment]
@dgsudharsan Please add me as reviewer.

**Comment by dgsudharsan** [comment]
> @dgsudharsan Please add me as reviewer.

Hi @ridahanif96  I am unable to add you as reviewer in the PRs

**Comment by ridahanif96** [comment]
> 

@dgsudharsan do you know who can help with this? , thanks.

**Comment by dgsudharsan** [comment]
> > 
> 
> @dgsudharsan do you know who can help with this? , thanks.

@zhangyanzhao Can you please help?

**Comment by dgsudharsan** [comment]
/azpw.run Azure.sonic-utilities

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by ridahanif96 — APPROVED** [review]

**Comment by prgeor** [comment]
> > > 
> > 
> > 
> > @dgsudharsan do you know who can help with this? , thanks.
> 
> @zhangyanzhao Can you please help?

@dgsudharsan reviewing...

**Comment by prgeor** [comment]
@dgsudharsan lets first align on the HLD

File: scripts/intfutil [hunk 1]
```diff
@@ -478,7 +491,7 @@ class IntfStatus(object):
                                 appl_db_port_status_get(self.db, key, PORT_LANES_STATUS),
                                 port_oper_speed_get(self.db, key),
                                 appl_db_port_status_get(self.db, key, PORT_MTU_STATUS),
-                                appl_db_port_status_get(self.db, key, PORT_FEC),
+                                port_oper_fec_get(self.db, key),
```
Comment (prgeor) [inline scripts/intfutil:494]: @dgsudharsan as discussed, lets not change this behavior. Existing 'FEC' column would continue to show whatever is configured in CONFIG_DB (i.e same as in APPL_DB).

**Review by ridahanif96 — APPROVED** [review]

File: scripts/intfutil [hunk 2]
```diff
@@ -806,6 +806,63 @@ class IntfLinkTrainingStatus(object):
...
+        #
+        for i in self.appl_db_keys:
+            key = re.split(':', i, maxsplit=1)[-1].strip()
+            if key in self.front_panel_ports_list:
+                if self.multi_asic.skip_display(constants.PORT_OBJ, key):
+                    continue
+                admin_fec = appl_db_port_status_get(self.db, key, PORT_FEC)
+                oper_fec = self.db.get(self.db.STATE_DB, PORT_STATE_TABLE_PREFIX + key, PORT_FEC)
+                oper_status = self.db.get(self.db.APPL_DB, PORT_STATUS_TABLE_PREFIX + key, PORT_OPER_STATUS)
+                if oper_status != "up" or oper_fec is None:
```
[the remark this answers is not in the document]
Comment (dgsudharsan) [reply scripts/intfutil:860]: When oper status is down, the status will be N/A. If the oper status is up, we will display whatever is there in the state_db. If the SAI doesn't support fetching oper fec, orchagent will update fec in state_db as N/A

[the remark this answers is not in the document]
Comment (prgeor) [reply scripts/intfutil:860]: @dgsudharsan misunderstood, line 858 return None on failure and its not 'None' FEC return value

**Review by prgeor — APPROVED** [review]

**Pull request merged**